### PR TITLE
Adds stage badge

### LINKED ISSUE #4872: Add-on stage support.
Addon can have follow stages: stable, experimental, deprecated. It's up to frontend to make a decision what kind of add-on is show to user.

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds a new stage bage to the addon info tab to show what stage the addon is in.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #4872
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 1]
```diff
@@ -205,6 +210,18 @@ class HassioAddonInfo extends LitElement {
               `
             : ""}
           <div class="security">
+            <ha-label-badge
```
Comment (frenck) [inline hassio/src/addon-view/info/hassio-addon-info.ts:213]: I still think we should iconize it... 🤷‍♂️ 
As the text will never fit in the badge itself.

**Review by pvizeli — APPROVED** [review]

File: hassio/src/addon-view/info/hassio-addon-info.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:213]: Sure, we can do that, but I think that will be confusing.
I'll throw up some examples in a few mins.

Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:213]: image
image
image
image

Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:213]: If they are added to the dialog as well, that might work:
image

Comment (frenck) [reply hassio/src/addon-view/info/hassio-addon-info.ts:213]: I like that approach by including the mapping in the description!

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 2]
```diff
@@ -205,6 +215,18 @@ class HassioAddonInfo extends LitElement {
...
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info.ts:218]: Add a check if `stage` is defined

File: hassio/src/addon-view/info/hassio-addon-info.ts (hunk 1 above)
Comment (bramkragten) [reply hassio/src/addon-view/info/hassio-addon-info.ts:218]: Maybe we only have to show this if it isn't stable?

File: hassio/src/addon-view/info/hassio-addon-info.ts (hunk 2 above)
Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:218]: stage is allways defined, it falls back to stable as a default

File: hassio/src/addon-view/info/hassio-addon-info.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info.ts:218]: I think they always should show, regardless of the stage.

File: hassio/src/addon-view/info/hassio-addon-info.ts (hunk 2 above)
Comment (bramkragten) [reply hassio/src/addon-view/info/hassio-addon-info.ts:218]: Then your types are wrong

File: src/data/hassio/addon.ts [hunk 3]
```diff
@@ -46,6 +46,7 @@ export interface HassioAddonDetails extends HassioAddonInfo {
   auto_uart: boolean;
   icon: boolean;
   logo: boolean;
+  stage?: string;
```
Comment (bramkragten) [inline src/data/hassio/addon.ts:49]: Here you say it could be `undefined`

Comment (ludeeus) [reply src/data/hassio/addon.ts:49]: True that!

Comment (bramkragten) [inline src/data/hassio/addon.ts]:
```suggestion
  stage: "stable" | "experimental" | "deprecated";
```

**Pull request merged**